templates/deployments.yaml

```diff
@@ -843,8 +843,8 @@ spec:
             /bin/cp -aR /opt/stackstorm/virtualenvs/. /opt/stackstorm/virtualenvs-shared
       # System packs
       - name: st2-system-packs
-        image: "{{ template "imageRepository" . }}/st2actionrunner{{ template "enterpriseSuffix" . }}:{{ .Chart.AppVersion }}"
-        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        image: "{{ template "imageRepository" $ }}/st2actionrunner{{ template "enterpriseSuffix" $ }}:{{ $.Chart.AppVersion }}"
+        imagePullPolicy: {{ $.Values.image.pullPolicy }}
```

Comment:
Curious what is the trick with `.` vs `$.`?
I don't see this widely demonstrated in Helm examples or other charts. Does it fixes anything, what's the difference?

Reply:
Ref: https://helm.sh/docs/chart_template_guide/#variables

> The `range` function will “range over” (iterate through) a list. But now something interesting happens. Just like with sets the scope of `.`, so does a `range` operator.
> 
> However, there is one variable that is always global - `$` - this variable will always point to the root context. This can be very useful when you are looping in a range and need to know the chart’s release name.

The scope of the range block is `.Values.st2.packs.sensors`. So any of the values outside that scope need to be referred to with `$`. I'm open to any other "cleaner" solution.

Reply:
Makes sense, thanks for explanation!